sim_pipeline/galaxy_galaxy_lens_pop.py

```diff
@@ -132,29 +127,29 @@ def select_lens_at_random(self, **kwargs_lens_cut):
             if gg_lens.validity_test(**kwargs_lens_cut):
                 return gg_lens
 
-    def deflector_number(self):
+    def potential_deflector_number(self):
         """Number of potential deflectors (meaning all objects with mass that are being
         considered to have potential sources behind them)
 
         :return: number of potential deflectors
         """
-        return self._lens_galaxies.deflector_number()
+        return len(self._lens_galaxies)
```

Comment:
is this function each time going to re-compute a quantity? There are certain functions that need repeated access to this number such that if this number is saved, can avoid unnecessary calculations

Reply:
I think here, len(self._lens_galaxies) will function the same way as self._lens_galaxies.deflector_number() as both methods are just calling a precomputed number that has been stored in the _lens_galaxies object!

Reply:
sounds good! not so used to overwrite class methods but good coding

Reply:
@padma18-vb  I reconsidered this point. I am not such a fan of overwriting __len__ functions. I am also not sure whether len() would work anymore if not defined for a class. Also, for backwards compatibility, having the function as it was does not hurt. It's mainly about backwards compatibility and it seems this change would not really be necessary to add your functionality